packages/compass-import-export/src/modules/export.js

```diff
@@ -383,10 +383,10 @@ export const openExport = ({
       dataService: { dataService },
     } = getState();
     try {
-      const count = isAggregation
-        ? null
-        : maybeCount ??
-          (await fetchDocumentCount(dataService, namespace, query));
+      const count =
+        !isAggregation && !maybeCount
+          ? await fetchDocumentCount(dataService, namespace, query)
+          : maybeCount;
```

Comment:
`!maybeCount` will be `true` in case when count is zero, nullish coalescing will handle this case. Maybe something like this to be a bit more explicit about our intended order (if count passed, use it, otherwise if not aggregation fetch results, otherwise `null`)?

```suggestion
      const count =
        maybeCount ??
        (!isAggregation
          ? await fetchDocumentCount(dataService, namespace, query)
          : null);
```